### PR TITLE
Adds pro/contra-speech feature to los

### LINKED ISSUE #191: Set pro/contra list-of-speakers
Adapt the new feature of OS3 to OS4: See [OS3#6023](https://github.com/OpenSlides/OpenSlides/pull/6023)

There is some work to do in the client and round about five new fields (for the models.yml and database).

### PR DESCRIPTION
**Pull request by GabrielInTheWorld**
As [OS3](https://github.com/OpenSlides/OpenSlides/pull/6023)

Fixes #191

### CODE REVIEW

**Comment by FinnStutzenstein** [comment]
Hi, please have a look at https://github.com/OpenSlides/OpenSlides/pull/6038 (and the linked issue)

File: client/src/app/core/repositories/event-management/meeting-settings-definition.ts [hunk 1]
```diff
@@ -354,6 +354,24 @@ export const meetingSettings: SettingsGroup[] = [
                         label: 'List of speakers is initially closed',
                         type: 'boolean',
                         default: false
+                    },
```
Comment (FinnStutzenstein) [inline client/src/app/core/repositories/event-management/meeting-settings-definition.ts:357]: The order was changed in the final PR (also check the wording)

File: client/src/app/core/repositories/agenda/speaker-repository.service.ts [hunk 2]
```diff
@@ -84,4 +89,12 @@ export class SpeakerRepositoryService extends BaseRepositoryWithActiveMeeting<Vi
         };
         return this.sendActionToBackend(SpeakerAction.UPDATE, payload);
     }
+
+    public setProContraSpeech(speaker: ViewSpeaker, isProSpeech: boolean): Promise<void> {
```
Comment (FinnStutzenstein) [inline client/src/app/core/repositories/agenda/speaker-repository.service.ts:93]: See my comment: this has changed a bit.

File: client/src/app/shared/components/list-of-speakers-content/list-of-speakers-content.component.ts [hunk 3]
```diff
@@ -84,6 +89,10 @@ export class ListOfSpeakersContentComponent extends BaseModelContextComponent im
         return !this.onlyPresentUsers || this.currentUser.isPresentInMeeting();
     }
 
+    public get showSpeakersOrderNote(): boolean {
```
Comment (FinnStutzenstein) [inline client/src/app/shared/components/list-of-speakers-content/list-of-speakers-content.component.ts:92]: See comment about the new fix-PR

**Review by FinnStutzenstein — APPROVED** [review]

**Review by tsiegleauq — APPROVED** [review]
Looks good. Works without issues on the client. Server seems not ready yet, so cannot be merged from my side

File: client/src/app/site/assignments/components/assignment-detail/assignment-detail.component.html [hunk 4]
```diff
@@ -148,15 +148,20 @@ <h3>{{ 'Candidates' | translate }}</h3>
...
-                                <button
-                                    mat-icon-button
-                                    matTooltip="{{ 'Remove candidate' | translate }}"
-                                    (click)="removeCandidate(item)"
-                                >
-                                    <mat-icon>clear</mat-icon>
-                                </button>
-                            </span>
+                            <div class="single-candidate-line">
+                                <ng-container *ngIf="hasPerms('manage')">
```
Comment (tsiegleauq) [inline client/src/app/site/assignments/components/assignment-detail/assignment-detail.component.html:152]: This was a bug in the original code. Move "hasPerms" to the `remove-candidate` button

**Review by tsiegleauq — CHANGES_REQUESTED** [review]

**Comment by FinnStutzenstein** [comment]
Please have a look into https://github.com/OpenSlides/OpenSlides/wiki/speaker.update and the models.yml

There is now just one field `speech_state` which is a combination of `pro_speech` and `marked` in OS3.

**Comment by FinnStutzenstein** [comment]
You may build upon https://github.com/OpenSlides/openslides-backend/pull/740

**Comment by tsiegleauq** [comment]
could directly integrate the following changes:
https://github.com/OpenSlides/OpenSlides/pull/6073/files

**Comment by GabrielInTheWorld** [comment]
@tsiegleauq I have reworked this. You can review it again.

**Review by tsiegleauq — APPROVED** [review]
perfect. Rebase and merge

**Comment by tsiegleauq** [comment]
hm I think I broke something.

**Comment by tsiegleauq** [comment]
ok that should do it.

**Pull request merged**